libs/utils/utils.js

```diff
@@ -686,3 +686,41 @@ export function loadLana(options = {}) {
   window.addEventListener('error', lanaError);
   window.addEventListener('unhandledrejection', lanaError);
 }
+
+/**
+* Searches for Japanese text in headings and applies a smart word-breaking algorithm by inserting
+* <wbr> between semantic blocks. This allows browsers to break japanese sentences correctly.
+*/
+export async function wordBreakJapanese() {
+  const { miloLibs, codeRoot, locale } = getConfig();
+
+  if (locale.ietf !== 'ja-JP') {
+    return;
+  }
+
+  const base = miloLibs || codeRoot;
+
+  // The thresould value to control word break granularity for long semantic blocks.
+  const WORD_SEG_THRES = 8;
+  const { loadDefaultJapaneseParser } = await import(`${base}/deps/budoux-index-ja.min.js`);
+  const parser = loadDefaultJapaneseParser();
+  document.querySelectorAll('h1, h2, h3, h4, h5').forEach((el) => {
+    // apply budoux to headings
+    parser.applyElement(el, { wordSegThres: WORD_SEG_THRES });
+  });
+
+  const BalancedWordWrapper = (await import(`${base}/deps/./bw2.min.js`)).default;
```

Comment:
@vhargrave 
Should we use camelCase even if it is a class? just make sure. I will follow your guideline.

>  Just ${base}/deps/bw2.min.js is ok.
Sorry typo

Reply:
If you're creating an **instance** of a class then camelCase (because it's in a variable). If you're **defining** a class then UpperCase is appropriate.

Reply:
@vhargrave Both are almost the same. But we should go with `balancedWordWrapper`, right?
```
import BalancedWordWrapper from './filename'
```

```
// NG
const BalancedWordWrapper =  await import('./filename').default
// OK
const balancedWordWrapper =  await import('./filename').default
```

Reply:
ah lol, I see your point . No, that looks awful.
Go with what you originally had - sorry!